ps2-decompile/attach.c

```diff
@@ -3,188 +3,170 @@
-                                            } else {
-                                                printf("sceUsbdOpenPipe failed(2)\n");
-                                                unit_free(g);
-                                                result = -1;
-                                            }
-                                        } else {
-                                            printf("sceUsbdOpenPipe failed(1)\n");
-                                            result = -1;
-                                        }
-                                    } else {
-                                        printf("unit_alloc failed\n");
-                                        result = -1;
-                                    }
-                                } else {
-                                    printf("wrong payload size %d\n", maxPacketSize);
-                                    result = -1;
-                                }
-                            } else {
-                                result = -1;
-                            }
-                        } else {
-                            result = -1;
-                        }
-                    } else {
-                        printf("sceUsbdScanStaticDescriptor (HID(0x21)) failed\n");
-                        result = -1;
-                    }
-                } else {
-                    printf("sceUsbdScanStaticDescriptor (ENDPOINT) failed\n");
-                    result = -1;
-                }
-            } else {
-                printf("wrong bNumEndpoints : %d\n", idesc->bNumEndpoints);
-                result = -1;
-            }
-        } else {
-            printf("sceUsbdScanStaticDescriptor (INTERFACE) failed\n");
-            result = -1;
-        }
-    } else {
+    if ((cdesc = sceUsbdScanStaticDescriptor(devId, 0, USB_DT_CONFIG)) == NULL) {
         printf("sceUsbdScanStaticDescriptor (CONFIGURATION) failed\n");
-        result = -1;
+        return -1;
+    }
+
+    if ((idec = sceUsbdScanStaticDescriptor(devId, cdesc, USB_DT_INTERFACE)) == NULL) {
+        printf("sceUsbdScanStaticDescriptor (INTERFACE) failed\n");
+        return -1;
+    }
+
+    if (idesc->bNumEndpoints != 1) {
+        printf("wrong bNumEndpoints : %d\n", idesc->bNumEndpoints);
+        return -1;
     }
-    return result;
+
+    if ((edesc = sceUsbdScanStaticDescriptor(devId, idesc, USB_DT_ENDPOINT)) == NULL) {
+        printf("sceUsbdScanStaticDescriptor (ENDPOINT) failed\n");
+        return -1;
+    }
+
+    if ((hdesc = sceUsbdScanStaticDescriptor(devId, 0, 33)) == NULL) {
+        printf("sceUsbdScanStaticDescriptor (HID(0x21)) failed\n");
+        return -1;
+    }
+
+    if ((edesc->bEndpointAddress & 0x80) == 0) {
+        return -1;
+    }
+
+    if ((edesc->bmAttributes & 3) != 3) {
+        return -1;
+    }
+
+    if ((maxPacketSize = edesc->wMaxPacketSizeLB | (edesc->wMaxPacketSizeHB << 8)) >= 9) {
```

Comment:
```suggestion
    int maxPacketSize = edesc->wMaxPacketSizeLB | (edesc->wMaxPacketSizeHB << 8;
    if (maxPacketSize >= 9) {
```

Reply:
i believe the current changes resolve this.